app/models/authentication_provider_github.rb

```diff
@@ -0,0 +1,20 @@
+class AuthenticationProviderGithub < ApplicationRecord
+  belongs_to :user
+
+  def self.find_or_create_user_from_auth_hash(auth)
+    authentication_provider_github = AuthenticationProviderGithub.eager_load(:user).find_by(uid: auth.uid)
+    return authentication_provider_github.user if authentication_provider_github
+
+    create_user_from_auth_hash(auth)
+  end
+
+  def self.create_user_from_auth_hash(auth)
+    user = User.new(name: auth["info"]["nickname"], role: "participant")
+    auth_provider = AuthenticationProviderGithub.new(uid: auth["uid"], user: user)
+    ApplicationRecord.transaction do
```

Comment:
NITS: we can remove `ApplicationRecord.` here since it's a class method.

Reply:
This is my habit.